src/Subscription/Lookup/Lookup.php

```diff
@@ -0,0 +1,189 @@
+
+        $self->criteria = $self->criteria->add(
+            new EventsCriterion(
+                array_values(
+                    array_map(
+                        function (string $event): string {
+                            if ($this->eventRegistry && $this->eventRegistry->hasEventClass($event)) {
+                                return $this->eventRegistry->eventName($event);
+                            }
+
+                            return $event;
+                        },
+                        $events,
+                    ),
+                ),
+            ),
+        );
+
+        return $self;
+    }
+
+    public function stream(string|null $stream = null): self
+    {
+        $self = clone $this;
+
+        if ($stream === null) {
+            $self->criteria = $self->criteria->remove(StreamCriterion::class);
+        } else {
+            $self->criteria = $self->criteria->add(new StreamCriterion($stream));
+        }
+
+        return $self;
+    }
+
+    public function currentStream(): self
+    {
+        $self = clone $this;
+
+        $stream = $self->currentMessage->header(StreamNameHeader::class)->streamName;
+        $self->criteria = $self->criteria->add(new StreamCriterion($stream));
+
+        return $self;
+    }
+
+    public function aggregateName(string|null $aggregateName = null): self
+    {
+        $self = clone $this;
+
+        if ($aggregateName === null) {
+            $self->criteria = $self->criteria->remove(AggregateNameCriterion::class);
+        } else {
+            $self->criteria = $self->criteria->add(new AggregateNameCriterion($aggregateName));
+        }
+
+        return $self;
+    }
+
+    public function aggregateId(string|null $aggregateId = null): self
+    {
+        $self = clone $this;
+
+        if ($aggregateId === null) {
+            $self->criteria = $self->criteria->remove(AggregateIdCriterion::class);
+        } else {
+            $self->criteria = $self->criteria->add(new AggregateIdCriterion($aggregateId));
+        }
+
+        return $self;
+    }
+
+    public function currentAggregate(): self
+    {
+        $self = clone $this;
+
+        $aggregateHeader = $self->currentMessage->header(AggregateHeader::class);
+        $self->criteria = $self->criteria
+            ->add(new AggregateNameCriterion($aggregateHeader->aggregateName))
+            ->add(new AggregateIdCriterion($aggregateHeader->aggregateId));
+
+        return $self;
+    }
+
+    public function forward(): self
+    {
+        $self = clone $this;
+
+        $self->backwards = false;
+
+        return $self;
+    }
+
+    public function backwards(): self
+    {
+        $self = clone $this;
+
+        $self->backwards = true;
+
+        return $self;
+    }
+
+    public function fetchAll(): Stream
+    {
+        return $this->store->load(
+            $this->criteria->add(
+                new ToIndexCriterion($this->currentMessage->header(IndexHeader::class)->index),
```

Comment:
this could already happen in the constructor, so then it never happens this will be missed, no?